presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -332,8 +335,44 @@ else if (javaType == Slice.class) {
         @Override
         protected Object visitDereferenceExpression(DereferenceExpression node, Object context)
         {
-            // Dereference is never a Symbol
-            return node;
+            Type type = expressionTypes.get(node.getBase());
+            // if there is no type for the base of Dereference, it must be QualifiedName
+            if (type == null) {
+                return node;
+            }
```

Comment:
What if `type` is not-null and not a `RowType`, can that happen? I think it could if you have a query like:
```
SELECT t.c
FROM t
```
Assuming c is a bigint